pkg/cni/cilium/cilium.go

```diff
@@ -353,15 +355,24 @@ func ValidateValuesUpdate(newValues, oldValues map[string]any, fieldPath *field.
 	}
 	allErrs = append(allErrs, validateImmutableValues(newOperator, oldOperator, operPath, []string{
 		"securityContext",
-	})...)
+	}, []string{})...)
 
 	return allErrs
 }
 
-func validateImmutableValues(newValues, oldValues map[string]any, fieldPath *field.Path, immutableValues []string) field.ErrorList {
+func validateImmutableValues(newValues, oldValues map[string]any, fieldPath *field.Path, immutableValues []string, excludedKeys []string) field.ErrorList {
 	allErrs := field.ErrorList{}
+	allowedValues := map[string]bool{}
+
 	for _, v := range immutableValues {
-		if fmt.Sprint(oldValues[v]) != fmt.Sprint(newValues[v]) {
+		for _, exclusion := range excludedKeys {
+			if strings.HasPrefix(exclusion, v) {
+				if excludedKeyExists(newValues, strings.Split(exclusion, ".")...) {
```

Comment:
tiny nit for future improvement: maybe we can store exclusions as both `string` and then `[]string` to not have to `strings.Split` the same constant values every time a webhook is called